microcosm_logging/tests/test_decorators.py

```diff
@@ -15,7 +19,43 @@ class TestClass(object):
     pass
 
 
+@logger
+class TestContextClass(object):
+
+    def function(self):
+        self.logger.info("success!")
+
+
 def test_using_class_logger_works():
     instance = TestClass()
 
     assert_that(instance.logger, is_(equal_to(getLogger(instance.__class__.__name__))))
+
+
+def test_using_context_logger_works():
+    instance = TestContextClass()
+    context_func = Mock(return_value={'some': 'context'})
+    wrapped = context_logger(context_func, instance.function, instance)
```

Comment:
From within the context_logger decorator (i.e. from the context that the decorated function is running in): https://github.com/globality-corp/microcosm-logging/pull/15/files#diff-6cb16950bb066336f47094d5b679db2bR61

The `context_func` tells the `conext_logger` how to look up context.